### ISSUE TITLE
`arguments` should not be allowed class static block

### ISSUE THREAD

**Issue by Zzzen** [microsoft/TypeScript#48152]
# Bug Report

### 🔎 Search Terms
arguments class static block error 

### 🕗 Version & Regression Information

- This is the behavior in every version I tried

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABAEQMwAoCUiDeBYAKEUQCcBTKEEpCAGwEMBnRxAFV0OOMantgg5EuxAO4ALGLTLoAjNnxDhxeiQDmIALZkwUFgHo9iRmLghaAE0RkSJOCU5KAvg+LOhbx0A)

### 💻 Code

```ts
function D3() {
  return class T {
    static {
      while(1) {
        arguments // should error
      }
    }
  }
}
```

### 🙁 Actual behavior

no error reported.

### 🙂 Expected behavior
a syntax error should be reported.

![image](https://user-images.githubusercontent.com/6630042/157023408-3078f8d1-2ab9-496e-8639-8476d44456a8.png)

related to #44269
